dali/python/nvidia/dali/pipeline.py

```diff
@@ -1255,12 +1243,61 @@ def _prefetch(self):
         if not self._pipe:
             raise RuntimeError("The pipeline was destroyed.")
         self._schedule_py_workers()
-        if self._exec_separated:
-            self._fill_separated_queues()
-        else:
-            for _ in range(self._prefetch_queue_depth):
-                self._run_once()
+
+        # We probably need some benchmarking before we remove this code path
+        if not self._exec_separated:
+            self._legacy_interleaved_prefetch()
```

Comment:
I have a concern/question here. It mainly impacts PES (but it may be relevant for memory usage in regular ES too).

First, do I read it right? For the most typical case (async pipelined), with the new way, we first produce `InputFeedCount = 2 * prefetch_queue_depth (cpu + gpu queues)` batches, fed them into external source (that is happy to accept batches irrespecitve of any prefetch_queues) and only then poke the executor to start?

If so, we increase the number of input batches to keep in external source twice, right? 

In the PES, we need to adjust for that explicitly, as the batches reside in a circular buffer of shared memory and are fed with no_copy into the backend external source op. Implementation wise it's easy, just pass the InputFeedCount to WorkerPool.from_groups call (instead of prefetch_queue_depth). 

The question is, if we want that: it seems to increase required memory and potentially slow down the start. What are the benefits?

Reply:
Maybe it wasn't clear, but, sure, I get we go with the legacy approach for now. The concern was more for the future consideration. Do you plan on getting rid of the legacy mode?

Reply:
Plan? No. Consider? Yes.